src/pages/index.js

```diff
@@ -0,0 +1,286 @@
+
+const previewModalImage = document.querySelector(".modal__image");
+const previewModalCaption = document.querySelector(".modal__caption");
+
+// Variables for delete functionality
+let selectedCard = null;
+let selectedCardId = null;
+let userId = null; // Store the current user's ID
+
+// Load initial data from API
+function loadInitialData() {
+  api
+    .getAppInfo()
+    .then(({ user, cards }) => {
+      console.log("API User Data:", user);
+      console.log("API Cards Data:", cards);
+      userId = user._id; // Store the current user's ID
+      profileName.textContent = user.name;
+      profileDescription.textContent = user.about;
+      profileAvatar.src = user.avatar;
+      cardsList.innerHTML = ""; // Clear existing cards
+      cards.forEach((item) => renderCard(item, "append"));
+    })
+    .catch((err) => console.error("Failed to load initial data:", err));
+}
+
+// Card creation function
+function getCardElement(data) {
+  console.log("Creating card with data:", data); // Debug: Log card data
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  // Check if the current user has liked the card
+  const isLiked = data.likes && data.likes.some((like) => like._id === userId);
+  console.log(`Card ${data._id} isLiked: ${isLiked}`); // Debug: Log initial like state
+  if (isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+
+  cardDeleteBtn.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data)
+  );
+  cardLikeBtn.addEventListener(
+    "click",
+    handleLikeClick.bind(null, cardLikeBtn, data)
+  );
+  cardImageEl.addEventListener("click", () => {
+    previewModalImage.src = data.link;
+    previewModalImage.alt = data.name;
+    previewModalCaption.textContent = data.name;
+    openModal(previewModal);
+  });
+
+  return cardElement;
+}
+
+function renderCard(item, method = "prepend") {
+  const cardElement = getCardElement(item);
+  cardsList[method](cardElement);
+}
+
+// Modal functions
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", closeModalByEscape);
+  modal.addEventListener("mousedown", closeModalByOverlayClick);
+  avatarContainer.classList.remove("profile__avatar-container--active");
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", closeModalByEscape);
+  modal.removeEventListener("mousedown", closeModalByOverlayClick);
+  avatarContainer.classList.remove("profile__avatar-container--active");
```

Comment:
Please, delete it


All modals should be opened only with `modal_opened`. You shouldn't us